src/modules/flow/json/json.c

```diff
@@ -0,0 +1,226 @@
+        if (sol_json_token_str_eq(&token_key, key, len))
+            return true;
+
+    return false;
+}
+
+static int
+json_object_key_process(struct sol_flow_node *node, struct sol_json_object_key *mdata)
+{
+    struct sol_json_scanner scanner;
+    struct sol_json_token value;
+    enum sol_json_type type;
+    struct sol_irange value_int = SOL_IRANGE_INIT();
+    struct sol_drange value_float = SOL_DRANGE_INIT();
+    struct sol_str_slice slice;
+    struct sol_blob *new_blob;
+    int r;
+
+    if (!mdata->key[0] || !mdata->json_object)
+        return 0;
+
+    sol_json_scanner_init(&scanner, mdata->json_object->mem,
+        mdata->json_object->size);
+
+    if (json_object_search_for_token(&scanner, mdata->key, &value)) {
+        type = sol_json_token_get_type(&value);
+        switch (type) {
+        case SOL_JSON_TYPE_OBJECT_START:
+            new_blob = create_sub_json_object(mdata->json_object, &value);
+            SOL_NULL_CHECK(new_blob, -errno);
+            r = sol_flow_send_json_object_packet(node,
+                SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__OBJECT,
+                new_blob);
+            sol_blob_unref(new_blob);
+            return r;
+        case SOL_JSON_TYPE_ARRAY_START:
+            new_blob = sol_blob_new(SOL_BLOB_TYPE_NOFREE,
+                mdata->json_object, value.start,
+                sol_json_token_get_size(&value));
+            SOL_NULL_CHECK(new_blob, -errno);
+            r = sol_flow_send_json_array_packet(node,
+                SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__ARRAY,
+                new_blob);
+            sol_blob_unref(new_blob);
+            return r;
+        case SOL_JSON_TYPE_TRUE:
+            return sol_flow_send_boolean_packet(node,
+                SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__BOOLEAN,
+                true);
+        case SOL_JSON_TYPE_FALSE:
+            return sol_flow_send_boolean_packet(node,
+                SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__BOOLEAN,
+                false);
+        case SOL_JSON_TYPE_NULL:
+            return sol_flow_send_empty_packet(node,
+                SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__NULL);
+        case SOL_JSON_TYPE_STRING:
+            sol_json_token_remove_quotes(&value);
+            slice = sol_json_token_to_slice(&value);
+            return sol_flow_send_string_slice_packet(node,
+                SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__STRING,
+                slice);
+        case SOL_JSON_TYPE_NUMBER:
+            r = sol_json_token_get_double(&value, &value_float.val);
+            SOL_INT_CHECK(r, < 0, r);
+            r = sol_flow_send_drange_packet(node,
+                SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__FLOAT,
+                &value_float);
+            SOL_INT_CHECK(r, < 0, r);
+
+            r = sol_json_token_get_int32(&value, &value_int.val);
+            if (r == -EINVAL) //Non an integer number
```

Comment:
I don't like the idea of sending the integer value if we have a float. I think this is confusing. If the field is a float number and user wants the int value, he should get the float value and convert it to int.